gnucash/gnome-utils/gnc-report-combo.c

```diff
@@ -0,0 +1,595 @@
+        gchar *report_guid;
+        gchar *report_name;
+        gtk_tree_model_get (model, &iter, RC_NAME, &report_name,
+                                          RC_GUID, &report_guid,
+                                          -1);
+
+        report = g_strconcat (report_guid, "/", report_name, NULL);
+        g_free (report_guid);
+        g_free (report_name);
+    }
+    return report;
+}
+
+void
+gnc_report_combo_set_active_guid_name (GncReportCombo *grc,
+                                       const gchar *guid_name)
+{
+    GncReportComboPrivate *priv;
+
+    g_return_if_fail (grc != NULL);
+    g_return_if_fail (GNC_IS_REPORT_COMBO(grc));
+
+    priv = GET_PRIVATE(grc);
+
+    if (guid_name && *guid_name)
+    {
+        gchar *guid = NULL;
+        gchar *name = g_strstr_len (guid_name, -1, "/");
+
+        if (name)
+        {
+            guid = g_strndup (guid_name, (name - guid_name));
+            gnc_report_combo_set_active (grc, guid, name + 1);
+        }
+        g_free (guid);
+    }
+}
+
+static void
+combo_changed_cb (GtkComboBox *widget, gpointer user_data)
+{
+    GncReportCombo        *grc = GNC_REPORT_COMBO(user_data);
+    GncReportComboPrivate *priv = GET_PRIVATE(grc);
+    GtkTreeIter            iter;
+
+    if (gtk_combo_box_get_active_iter (widget, &iter))
+    {
+        GtkTreeModel *model = gtk_combo_box_get_model (widget);
+        gchar *name, *guid;
+        gboolean missing;
+        gtk_tree_model_get (model, &iter, RC_NAME, &name,
+                                          RC_GUID, &guid,
+                                          RC_MISSING, &missing,
+                                          -1);
+        // set visibility of the warning image
+        gtk_widget_set_visible (priv->warning_image, missing);
+
+        if (!priv->block_signal)
+            g_signal_emit (grc, report_combo_signals [CHANGED], 0);
+
+        gtk_widget_queue_resize (GTK_WIDGET(widget));
+        g_free (name);
+        g_free (guid);
+    }
+}
+
+static void
+combo_popped_cb (GObject    *gobject,
+                 GParamSpec *pspec,
+                 gpointer    user_data)
+{
+    GncReportCombo        *grc = GNC_REPORT_COMBO(user_data);
+    GncReportComboPrivate *priv = GET_PRIVATE(grc);
+    gboolean popup_shown;
+
+    g_object_get (G_OBJECT(gobject), "popup-shown", &popup_shown, NULL);
+
+    priv->popup_shown = popup_shown;
+    g_object_notify (G_OBJECT(grc), "popup-shown");
+}
+
+void
+gnc_report_combo_refresh (GncReportCombo *grc)
+{
+    GncReportComboPrivate *priv;
+
+    g_return_if_fail (grc != NULL);
+    g_return_if_fail (GNC_IS_REPORT_COMBO(grc));
+
+    priv = GET_PRIVATE(grc);
+
+    priv->block_signal = TRUE;
+
+    update_report_list (grc);
+
+    if (!select_active (grc))
+        update_warning_tooltip (grc);
+
+    priv->block_signal = FALSE;
+}
+
+GtkWidget *
+gnc_report_combo_get_warning_image (GncReportCombo *grc)
```

Comment:
Unless I missed something, I don't see this function used anywhere. Why does it exist and why is it exposed in the public interface of this widget ?

Reply:
I added this at the start and as you pointed out never used so deleted.